itkwidgets/viewer.py

```diff
@@ -144,115 +149,143 @@ def __init__(
         self, ui_collapsed=True, rotate=False, ui="pydata-sphinx", **add_data_kwargs
     ):
         """Create a viewer."""
+        self.bg_jobs = bg.BackgroundJobManager()
         self.viewer_rpc = ViewerRPC(
             ui_collapsed=ui_collapsed, rotate=rotate, ui=ui, **add_data_kwargs
         )
+        self.queue = queue.Queue()
+        self.bg_thread = self.bg_jobs.new(self.queue_worker)
         api.export(self.viewer_rpc)
 
+    @property
+    def loop(self):
+        return asyncio.get_running_loop()
+
+    async def run_queued_requests(self):
+        self.viewer_rpc.event.wait()
+        while self.queue.qsize():
+            await asyncio.sleep(1)
```

Comment:
Replace with callback for setImage, etc., remove sleeps

Reply:
Updated, no longer sleeps. Instead uses events and queues to manage what methods are called when. All data setters and UI methods are queued to be called immediately after the viewer is created. Any data property setters are added to a deferred queue and called as soon as the data has been set.